### PR TITLE
Make fields in ThreadPool readonly/const where possible

### PR DESCRIPTION
**Pull request by svick**
Also remove unused usings

### CODE REVIEW

File: src/mscorlib/src/System/Threading/ThreadPool.cs [hunk 1]
```diff
@@ -145,7 +143,7 @@ internal class WorkStealingQueue
             private volatile int m_headIndex = START_INDEX;
             private volatile int m_tailIndex = START_INDEX;
 
-            private SpinLock m_foreignLock = new SpinLock(false);
+            private readonly SpinLock m_foreignLock = new SpinLock(false);
```
Comment (omariom) [inline src/mscorlib/src/System/Threading/ThreadPool.cs]: This field shouldn't be declared readonly as SpinLock is mutable struct.

Comment (svick) [inline src/mscorlib/src/System/Threading/ThreadPool.cs]: Thanks, fixed.

I wonder if there some reasonable way to automatically catch such errors in the future.

Comment (omariom) [inline src/mscorlib/src/System/Threading/ThreadPool.cs]: Probably an attribute and analyzer. It could check that no copies are made.

**Comment by danmosemsft** [comment]
@dotnet-bot test this please

**Comment by danmosemsft** [comment]
@dotnet-bot test osx x64 checked build and test ('java / Jenkins hit a remoting error that caused the build to fail.')

**Comment by danmosemsft** [comment]
@kouvel please review and merge if appropriate.

**Comment by kouvel** [comment]
Thanks for cleaning this up!

**Pull request merged**